src/convertToObject.js

```diff
@@ -3,16 +3,32 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with
+ * styles (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related
+ * CSS properties (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitedProperty = sourceString.split(';');
+  const propertiesList = splitedProperty.map(property => {
+    let [key, value] = property.split(':');
+
+    key = key.replace(/\n| /g, '');
+    value && (value = value.trim());
+
+    if (key === '') {
+      return;
+    }
+
+    return { [key]: value };
```

Comment:
You return undefined or object in a map. This will result in smth like: 
`[undefined, { color: 'red' }, undefined, { border: '1px solid' }]`

consider following common solution, and improve your code here:

```
const styleObject = {};
  
  const properties = sourceString
    .split(';')
    .map((elem) => elem.trim())
    .filter((elem) => elem.length > 0)
    .map((elem) => elem.split(':'))
    .map((elem) => elem.map((n) => n.trim()));
  properties.forEach(([property, value]) => (styleObject[property] = value));
  return styleObject;
```